stdlib/LinearAlgebra/test/matmul.jl

```diff
@@ -172,119 +172,152 @@ end
+        @test dot(v, a) ≈ 67.0
+        @test dot(Vector(v), Vector(v)) ≈ 83.0
+        @test dot(v, v) ≈ 83.0
+    end
+end
+
+@testset "Complex matrix x real MatOrVec etc (issue #29224)" for T1 in (Float32, Float64)
+    for T2 in (Float32, Float64)
+        for arg1_real in (true, false)
+            @testset "Combination $T1 $T2 $arg1_real $arg2_real" for arg2_real in (true, false)
+                A0 = reshape(Vector{T1}(1:25), 5, 5) .+
+                     (arg1_real ? 0 : 1im * reshape(Vector{T1}(-3:21), 5, 5))
+                A = view(A0, 1:2, 1:2)
+                B = Matrix{T2}([1.0 3.0; -1.0 2.0]) .+
+                    (arg2_real ? 0 : 1im * Matrix{T2}([3.0 4; -1 10]))
+                AB_correct = copy(A) * B
+                AB = A * B  # view times matrix
                 @test AB ≈ AB_correct
-                A1 = view(A0,:,1:2)  # rectangular view times matrix
-                @test A1*B ≈ copy(A1)*B
-                B1 = view(B,1:2,1:2);
-                AB1 = A*B1; # view times view
+                A1 = view(A0, :, 1:2)  # rectangular view times matrix
+                @test A1 * B ≈ copy(A1) * B
+                B1 = view(B, 1:2, 1:2)
+                AB1 = A * B1 # view times view
                 @test AB1 ≈ AB_correct
-                x = Vector{T2}([1.0;10.0]) .+ (arg2_real ? 0 : 1im*Vector{T2}([3;-1]))
-                Ax_exact = copy(A)*x
-                Ax = A*x  # view times vector
+                x = Vector{T2}([1.0; 10.0]) .+ (arg2_real ? 0 : 1im * Vector{T2}([3; -1]))
+                Ax_exact = copy(A) * x
+                Ax = A * x  # view times vector
                 @test Ax ≈ Ax_exact
-                x1 = view(x,1:2)
-                Ax1 = A*x1  # view times viewed vector
+                x1 = view(x, 1:2)
+                Ax1 = A * x1  # view times viewed vector
                 @test Ax1 ≈ Ax_exact
-                @test copy(A)*x1 ≈ Ax_exact # matrix times viewed vector
+                @test copy(A) * x1 ≈ Ax_exact # matrix times viewed vector
                 # View times transposed matrix
-                Bt = transpose(B);
-                @test A*Bt ≈ A*copy(Bt)
+                Bt = transpose(B)
+                @test A * Bt ≈ A * copy(Bt)
             end
         end
     end
 end
 
+@testset "real matrix x complex vec" begin
+    n = 4
+
+    _matmulres(M, v) = [mapreduce(*, +, row, v) for row in eachrow(M)]
+    testmatmul(M, v) = @test M * v ≈ _matmulres(M, v)
+
+    @testset for T in (Float32, Float64)
+        M1 = reshape(Vector{T}(1:n^2), n, n)
+        M2 = reinterpret(reshape, T, [T.((i, i + 1, i + 2, i + 3)) for i in 1:4])
+        v = Vector(T.(1:n)) .+ im .* Vector(T.(4 .+ (1:n)))
+
+        for M in (M1, M2)
+            M_view_cont = @view M[:, :]
+            v_view_cont = @view v[:]
+            for _M in (M, M_view_cont), _v in (v, v_view_cont)
+                testmatmul(_M, _v)
+            end
+
+            M_view_noncont = @view M[1:2:end, 1:2:end]
```

Comment:
One last suggestion:
```suggestion
            M_view_noncont = @view M[1:2, 1:2:end]
```
(I think there's no need to add more test to genetic fallback. So it would be good to make sure `stride(M_view_noncont, 1) == 1`)

Reply:
updated now